key/src/main/java/software/amazon/kms/key/ModelAdapter.java

```diff
@@ -1,23 +1,30 @@
 package software.amazon.kms.key;
 
 import com.amazonaws.util.StringUtils;
+
+import software.amazon.awssdk.services.kms.model.CustomerMasterKeySpec;
 import software.amazon.awssdk.services.kms.model.KeyUsageType;
 
 public class ModelAdapter {
-    protected static final String DEFAULT_DESCRIPTION = "";
-    protected static final Boolean DEFAULT_ENABLED = true;
-    protected static final Boolean DEFAULT_ENABLE_KEY_ROTATION = false;
+    private static final String DEFAULT_DESCRIPTION = "";
+    private static final Boolean DEFAULT_ENABLED = true;
+    private static final Boolean DEFAULT_ENABLE_KEY_ROTATION = false;
+    private static final String DEFAULT_KEY_USAGE = KeyUsageType.ENCRYPT_DECRYPT.toString();
+    private static final String DEFAULT_KEY_SPEC = CustomerMasterKeySpec.SYMMETRIC_DEFAULT.toString();
+
     public static ResourceModel setDefaults(final ResourceModel model) {
 
         final String description = model.getDescription();
         final Boolean enabled = model.getEnabled();
         final Boolean enabledKeyRotation = model.getEnableKeyRotation();
         final String keyUsage = model.getKeyUsage();
+        final String keySpec = model.getKeySpec();
 
         model.setDescription(StringUtils.isNullOrEmpty(description) ? DEFAULT_DESCRIPTION : description);
         model.setEnabled(enabled == null ? DEFAULT_ENABLED : enabled);
         model.setEnableKeyRotation(enabledKeyRotation == null ? DEFAULT_ENABLE_KEY_ROTATION : enabledKeyRotation);
```

Comment:
maybe for safety we should use:
`model.setEnableKeyRotation(enabledKeyRotation == null && model.getKeyUsage == DEFAULT_KEY_USAGE ? DEFAULT_ENABLE_KEY_ROTATION : enabledKeyRotation);`

Reply:
See my other comment about leaving enabledKeyRotation null.